components/net/indexeddb/engines/sqlite.rs

```diff
@@ -0,0 +1,513 @@
+                                            .map(|blob| serialize::decode(&blob.key).unwrap())));
+                            },
+                            Err(e) => {
+                                let _ = sender.send(err!(e));
+                            },
+                        }
+                    },
+                }
+            }
+        });
+        rx
+    }
+
+    // TODO: we should be able to error out here, maybe change the trait definition?
+    fn has_key_generator(&self, store_name: SanitizedName) -> bool {
+        HANDLE.block_on(async {
+            if let Some(model) = object_store_model::Entity::find()
+                .filter(object_store_model::Column::Name.eq(store_name.to_string()))
+                .one(&self.connection)
+                .await
+                .unwrap()
+            {
+                model.auto_increment
+            } else {
+                false
+            }
+        })
+    }
+
+    fn key_path(&self, store_name: SanitizedName) -> Option<KeyPath> {
+        HANDLE.block_on(async {
+            if let Some(model) = object_store_model::Entity::find()
+                .filter(object_store_model::Column::Name.eq(store_name.to_string()))
+                .one(&self.connection)
+                .await
+                .unwrap()
+            {
+                model
+                    .key_path
+                    .map(|key_path| bincode::deserialize(&key_path).unwrap())
+            } else {
+                None
+            }
+        })
+    }
+
+    fn create_index(
+        &self,
+        store_name: SanitizedName,
+        index_name: String,
+        key_path: KeyPath,
+        unique: bool,
+        multi_entry: bool,
+    ) -> Result<CreateObjectResult, Self::Error> {
+        HANDLE.block_on(async {
+            let object_store = match object_store_model::Entity::find()
+                .filter(object_store_model::Column::Name.eq(store_name.to_string()))
+                .one(&self.connection)
+                .await?
+            {
+                Some(model) => model,
+                None => {
+                    return Err(Self::Error::Custom("No such store".to_string()));
+                },
+            };
+            if object_store_index_model::Entity::find()
+                .filter(
+                    object_store_index_model::Column::Name
+                        .eq(index_name.to_string())
+                        .and(object_store_index_model::Column::ObjectStoreId.eq(object_store.id)),
+                )
+                .one(&self.connection)
+                .await?
+                .is_some()
+            {
+                return Ok(CreateObjectResult::AlreadyExists);
+            }
+            let model = object_store_index_model::ActiveModel {
+                id: Default::default(),
+                object_store_id: Set(object_store.id),
+                name: Set(index_name),
+                key_path: Set(bincode::serialize(&key_path).unwrap()),
```

Comment:
Should we use the serialize module instead?

Reply:
I don't think so, the serialize module is just to ensure that keys get properly sorted and so that key ranges work.